### PR TITLE
[Target] A better way of notifying ASTContext(s). NFCI.

### PR DESCRIPTION
**Pull request by dcci**
Discussed with Adrian.

<rdar://problem/30398933>

### CODE REVIEW

**Comment by dcci** [comment]
@adrian-prantl I'm a little nervous about this one so I would appreciate if you can take a look.

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@swift-ci please test linux

**Comment by dcci** [comment]
@swift-ci please test macOS and merge

**Comment by dcci** [comment]
@swift-ci please test macos

File: source/Target/Target.cpp [hunk 1]
```diff
@@ -1645,15 +1645,25 @@ void Target::ModulesDidLoad(ModuleList &module_list) {
...
+          llvm::dyn_cast_or_null<SwiftASTContext>(type_system);
+      if (!swift_ast_ctx)
+        return true;
       swift_ast_ctx->ModulesDidLoad(module_list);
+      return true;
+    };
+    m_scratch_type_system_map.ForEach(notify_callback);
+
+    // This is a DenseMap, but we're fine iterating over it because
+    // it doens't matter in which order we notify the ASTContext(s).
```
Comment (adrian-prantl) [inline source/Target/Target.cpp:1661]: doesn't :-)

**Review by adrian-prantl — APPROVED** [review]
Yes, this is precisely what the function should have been!

**Pull request merged**